sdk/core/Azure.Core/src/FixedSizedBufferWriter.cs

```diff
@@ -0,0 +1,187 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Buffers;
+using System.Diagnostics;
+
+namespace Azure.Core
+{
+    /// <summary>
+    /// Represents a heap-based, array-backed output sink into which <typeparam name="T"/> data can be written.
+    /// </summary>
+    public sealed class ArrayBufferWriter<T> : IBufferWriter<T>
```

Comment:
I don't know if it's worth making this a public type. The type ships in the BCL and will be obsolete at some point. I am thinking we should do what @weshaggard suggested: create folder in the repo and put all shared sources there, and then share this type and HashCode as sources sucked into each project.

Reply:
Done.